### PR TITLE
[Bugfix] Fix NPE for automatic feedback as the build can fail

### PR DESCRIPTION
**Pull request by fde312**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Added screenshots

### Motivation and Context

When the build fails of a student's submission no automatic feedback is created. A null check was missing in the current implementation.

### Description

- Added null check when setting automatic feedback

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercise and participate and make your build fail!
4. Assess programming exercise and see that the build fail participation is loaded and you can submit your assessment as usual.

### Test Coverage

Nothing changed

### Screenshots

image

image

### CODE REVIEW

**Review by MaisiKoleni — DISMISSED** [review]
Code looks good to me.

**Review by kloessst — APPROVED** [review]
Good catch! Code LGTM.

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts [hunk 1]
```diff
@@ -371,7 +371,7 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
 
     private handleFeedback(): void {
         // Setup automatic feedback
-        this.automaticFeedback = this.automaticResult?.feedbacks!;
+        this.automaticFeedback = this.automaticResult?.feedbacks! ?? [];
```
Comment (krusche) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts:374]: does this really work? I guess you would need to leave out the `!` operator after `feedbacks`

Comment (krusche) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts:374]:
```suggestion
        this.automaticFeedback = this.automaticResult?.feedbacks ?? [];
```

Comment (fde312) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts:374]: you are right, this is not needed and can be removed. Good catch! 👍

Comment (kloessst) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts]: I think it still works but removing the ! is cleaner.

**Review by kloessst — APPROVED** [review]
Re-approve

**Review by krusche — APPROVED** [review]

**Pull request merged**